scripts/constituents.py

```diff
@@ -525,25 +523,68 @@ def write_host_routines(cap, host, reg_funcname, init_funcname, num_const_funcna
             cap.write(f"num_suite_consts = {funcname}({errvar_str})", 2)
             cap.write("num_consts = num_consts + num_suite_consts", 2)
         # end for
-        cap.write("if ({} == 0) then".format(herrcode), 2)
+        # Check for dynamic constituent routines
+        if len(dyn_const_dict) > 0:
+            cap.comment("Add in dynamic constituents", 2)
+            for idx, scheme in enumerate(sorted(dyn_const_dict)):
+                cap.write(f"if ({herrcode} == 0) then", 2)
+                cap.write(f"call {dyn_const_dict[scheme]}(dyn_const_prop_{idx}, {obj_err_callstr})", 3)
+                cap.write(f"num_dyn_consts = num_dyn_consts + size(dyn_const_prop_{idx})", 3)
+                cap.write("end if", 2)
+            # end for
+            cap.write("num_consts = num_consts + num_dyn_consts", 2)
+            cap.comment("Pack dynamic_constituents array", 2)
+            cap.write(f"allocate(dynamic_constituents(num_dyn_consts), stat={herrcode})", 2)
+            cap.write(f"if ({herrcode} /= 0) then", 2)
+            cap.write(f"{herrmsg} = 'failed to allocate dynamic_constituents'", 3)
+            cap.write("else", 2)
+            cap.write("index_start = 1", 3)
+            for idx, scheme in enumerate(sorted(dyn_const_dict)):
+                cap.write(f"do index = 1, size(dyn_const_prop_{idx}, 1)", 3)
+                cap.write(f"dynamic_constituents(index + index_start - 1) = dyn_const_prop_{idx}(index)", 4)
+                cap.write("end do", 3)
+                cap.write(f"index_start = size(dyn_const_prop_{idx}, 1) + 1", 3)
+                cap.write(f"deallocate(dyn_const_prop_{idx})", 3)
+            # end for
+            cap.write("end if", 2)
```

Comment:
I think (?) we can eliminate some of the ones floating around here:

```suggestion
            cap.write("index_start = 0", 3)
            for idx, scheme in enumerate(sorted(dyn_const_dict)):
                cap.write(f"do index = 1, size(dyn_const_prop_{idx}, 1)", 3)
                cap.write(f"dynamic_constituents(index + index_start) = dyn_const_prop_{idx}(index)", 4)
                cap.write("end do", 3)
                cap.write(f"index_start = size(dyn_const_prop_{idx}, 1)", 3)
                cap.write(f"deallocate(dyn_const_prop_{idx})", 3)
            # end for
            cap.write("end if", 2)
```

Reply:
oooh. nice. updated!